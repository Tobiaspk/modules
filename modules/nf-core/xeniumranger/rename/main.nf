process XENIUMRANGER_RENAME {
    tag "$meta.id"
    label 'process_high'

    container "nf-core/xeniumranger:3.0.1"

    input:
    path(xenium_bundle)

    output:
    path("outs/**"), emit: outs
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    // Exit if running this module with -profile conda / -profile mamba
    if (workflow.profile.tokenize(',').intersect(['conda', 'mamba']).size() >= 1) {
        error "XENIUMRANGER_RENAME module does not support Conda. Please use Docker / Singularity / Podman instead."
    }
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    def region_name = region_name ? "--region_name=\"${region_name}\"": ""
    def cassette_name = cassette_name ? "--cassette_name=\"${cassette_name}\"": ""

    """
    xeniumranger rename \\
        --id="${prefix}" \\
        --xenium-bundle="${xenium_bundle}" \\
        ${region_name} \\
        ${cassette_name} \\
        --localcores=${task.cpus} \\
        --localmem=${task.memory.toGiga()} \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        xeniumranger: \$(xeniumranger -V | sed -e "s/xeniumranger-/- /g")
    END_VERSIONS
    """

    stub:
    // Exit if running this module with -profile conda / -profile mamba
    if (workflow.profile.tokenize(',').intersect(['conda', 'mamba']).size() >= 1) {
        error "XENIUMRANGER_RELABEL module does not support Conda. Please use Docker / Singularity / Podman instead."
    }
    """
    mkdir -p outs/
    touch outs/fake_file.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        xeniumranger: \$(xeniumranger -V | sed -e "s/xeniumranger-/- /g")
    END_VERSIONS
    """
}
