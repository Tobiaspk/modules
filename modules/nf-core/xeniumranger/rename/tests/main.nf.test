nextflow_process {

    name "Test Process XENIUMRANGER_RENAME"
    script "../main.nf"
    process "XENIUMRANGER_RENAME"

    tag "modules"
    tag "modules_nfcore"
    tag "xeniumranger"
    tag "xeniumranger/rename"

    test("xeniumranger relabel") {

        when {

            process {
                """
                input[0] = path(params.test_data['mus_musculus']['10xgenomics']['xeniumranger']['xeniumbundle'], checkIfExists: true)
                input[1] = path(params.test_data['mus_musculus']['10xgenomics']['xeniumranger']['gene_panel'], checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.versions,
                    process.out.outs.get(0).get(1).findAll { file(it).name !in [
                        'analysis_summary.html',
                        'cell_feature_matrix.h5',
                        'cell_feature_matrix.zarr.zip',
                        'barcodes.tsv.gz',
                        'features.tsv.gz',
                        'matrix.mtx.gz',
                        ]}
                    ).match()
                },
                { assert file(process.out.outs.get(0).get(1).find { file(it).name == 'analysis_summary.html' }).exists() },
                { assert file(process.out.outs.get(0).get(1).find { file(it).name == 'cell_feature_matrix.h5' }).exists() },
                { assert file(process.out.outs.get(0).get(1).find { file(it).name == 'cell_feature_matrix.zarr.zip' }).exists() },
                { assert file(process.out.outs.get(0).get(1).find { file(it).name == 'barcodes.tsv.gz' }).exists() },
                { assert file(process.out.outs.get(0).get(1).find { file(it).name == 'features.tsv.gz' }).exists() },
                { assert file(process.out.outs.get(0).get(1).find { file(it).name == 'matrix.mtx.gz' }).exists() }
            )
        }
    }

    test("xeniumranger relabel -stub") {

        options "-stub"

        when {

            process {
                """
                input[0] = path(params.test_data['mus_musculus']['10xgenomics']['xeniumranger']['xenium_bundle'], checkIfExists: true)
                input[1] = path(params.test_data['mus_musculus']['10xgenomics']['xeniumranger']['gene_panel'], checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}