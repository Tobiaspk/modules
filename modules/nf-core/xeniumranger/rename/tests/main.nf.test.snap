{
    "xeniumranger rename stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_run"
                        },
                        "fake_file.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,823917e7bc8f27cf314ef477fe7369eb"
                ],
                "outs": [
                    [
                        {
                            "id": "test_run"
                        },
                        "fake_file.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,823917e7bc8f27cf314ef477fe7369eb"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-22T15:23:09.592208"
    },
    "xeniumranger rename": {
        "content": [
            [
                "versions.yml:md5,823917e7bc8f27cf314ef477fe7369eb"
            ],
            [
                "analysis.zarr.zip:md5,1ffb1b86586fe6c80ce1676b79137785",
                "cell_boundaries.csv.gz:md5,8b4f2aa455a6fb14b2669a42db32ea7e",
                "cell_boundaries.parquet:md5,e55d6a7fbec336103994baad8c8e4a9a",
                "cells.csv.gz:md5,3cef2d7cc8cfba1d47bdb7c65c3d5d5f",
                "cells.parquet:md5,e1450c7eca3d7ce0d4911c95042b1303",
                "cells.zarr.zip:md5,556e47d5b14150239b10b2f801defa2b",
                "experiment.xenium:md5,bad9064b95722ad99eb66d1f0c5f1828",
                "gene_panel.json:md5,8890dd5fd90706e751554ac3fdfdedde",
                "metrics_summary.csv:md5,2c5412059595b11201bd9b11bc1661af",
                "morphology.ome.tif:md5,6b65fff28a38a001b8f25061737fbf9b",
                "morphology_focus_0000.ome.tif:md5,90e796ad634d14e62cf2ebcadf2eaf98",
                "nucleus_boundaries.csv.gz:md5,e417b6e293298870956d42c7106cbd0c",
                "nucleus_boundaries.parquet:md5,bacbfc3c2e956d899e1d8ccba5dd7c5e",
                "transcripts.parquet:md5,203cb05ee7689805cc505ebda9557551",
                "transcripts.zarr.zip:md5,807e63a2ef8340e085cd899507f45395"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-22T15:22:54.88242"
    }
}